### DISCUSSION TITLE
Flash an example of espressif with jtag programmer on ai deck board

### DISCUSSION THREAD

**Discussion by mvtkurd** [bitcraze/discussions#248] [Q&A]
Hi everybody,

I want to flash tcp [example](https://github.com/espressif/esp-idf/tree/v4.3.1/examples/protocols/sockets/tcp_client) of espressif on ai deck . I used these commands `$ source ~/path/to/idf/export.sh   $ idf.py build  ` as you referred in the esp32 repository and I built the binary file but I don't know how could I flash this binary file with ARM-USB-OCD-H jtag programmer  on the board. Thanks.

**Comment by evoggy** [comment]
Hi,

There's more information on how to build and flash the ESP32 via JTAG in the repository [readme](https://github.com/bitcraze/aideck-esp-firmware/blob/main/README.md#flash-using-the-builder-image). You can use the docker builder image for the flashing (you can do this even if you're not building the firmware with the builder):

```docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/builder /bin/bash -c "openocd -f interface/ftdi/olimex-arm-usb-tiny-h.cfg -f board/esp-wroom-32.cfg -c 'adapter_khz 20000' -c 'program_esp build/bootloader/bootloader.bin 0x1000 verify' -c 'program_esp build/aideck_esp.bin 0x10000 verify reset exit'"```

**Reply by mvtkurd** [reply]
@evoggy I got this error using the mentioned command: 

![Screenshot 2022-09-01 122643](https://user-images.githubusercontent.com/32674939/187892864-5389cf75-6fe1-4229-8700-24f521c04cc6.jpg)

**Reply by evoggy** [reply]
Ok, strange. It seems to identify the targets but it cannot reset it the ESP32. Is it just the AI deck connected to the Crazyflie? Could you try to push the connector a bit in case there might be a glitch?

**Reply by mvtkurd** [reply]
yes only aideck is attached. I've checked the connections with multimeter and seems everything is fine but for one time I got this error(attached picture) instead of previous one.


![2](https://user-images.githubusercontent.com/32674939/188097112-eeacd712-a2f1-4f79-9539-f1914fd6e192.jpg)

**Comment by knmcguire** [comment]
Just a question here as I have experienced this before: From which OS are you trying to flash the ESP? Ubuntu, Windows, WSL?

**Reply by mvtkurd** [reply]
I'm using Ubuntu. In fact , the bitcraze virtual machine.

**Reply by knmcguire** [reply]
Hmm interesting. I've had the same issue when I connected my programmer's usb to WSL which is also a virtual machine ... but on a native install of ubuntu I didn't had this problem. I wonder if the virtual machine USB is suitable for a programmer but I can't find anything online that supports this.

This requires more investigation probably. It's an annoying problem.

Do you perhaps have a different programmer like a j-link to try? or a computer with a native install of ubuntu?
